poly-commitment/src/commitment.rs

```diff
@@ -62,8 +62,8 @@ where
         F: FnMut(A) -> B,
         B: CanonicalDeserialize + CanonicalSerialize,
     {
-        let unshifted = self.unshifted.iter().map(|x| f(*x)).collect();
-        let shifted = self.shifted.map(f);
+        let unshifted = self.unshifted.iter().map(|x| f(x.clone())).collect();
```

Comment:
`self.unshifted.iter().cloned().map(f).collect()`?

Reply:
I'm not able to do this. If I write it this way, `f` gets moved, and so I can't use it for `shifted`.